core/trino-main/src/test/java/io/trino/operator/scalar/TestFailureFunction.java

```diff
@@ -22,25 +22,30 @@
 
 import static io.trino.SessionTestUtils.TEST_SESSION;
 import static io.trino.type.UnknownType.UNKNOWN;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 public class TestFailureFunction
         extends AbstractTestFunctions
 {
     private static final String FAILURE_INFO = JsonCodec.jsonCodec(FailureInfo.class).toJson(Failures.toFailure(new RuntimeException("fail me")).toFailureInfo());
 
-    @Test(expectedExceptions = RuntimeException.class, expectedExceptionsMessageRegExp = "fail me")
+    @Test
     public void testFailure()
     {
-        assertFunction("fail(json_parse('" + FAILURE_INFO + "'))", UNKNOWN, null);
+        assertThatThrownBy(() -> assertFunction("fail(json_parse('" + FAILURE_INFO + "'))", UNKNOWN, null))
+                .isInstanceOf(RuntimeException.class)
+                .hasMessageMatching("fail me");
```

Comment:
We can use `assertInvalidFunction()` here. 

(The original test is just one line, so it's was no problem though)

Reply:
You shouldn't delete `fail()` because this test is for `fail()` function. 
```java
assertInvalidFunction("fail(json_parse('" + FAILURE_INFO + "'))", GENERIC_USER_ERROR, "fail me");
```

Reply:
Yea, I believe that failed as well, but with a different error. Lemme post it here.

Reply:
Nvm, I wasn't passing the `GENERIC_USER_ERROR` parameter to `assertInvalidFunction()` when testing it with fail(). I will try the new code. Thanks for sharing it.